### PR TITLE
[CON-380] Migrate `SecondaryMessage` to TypeScript

### PR DESCRIPTION
**Pull request by beau-charman**
Addresses https://aesoponline.atlassian.net/browse/CON-380

### CODE REVIEW

File: src/components/SecondaryMessage/SecondaryMessage.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+            link: {
+              title: 'SecondaryMessage link title',
+              url: 'SecondaryMessage link url',
+            },
+          },
+        ]}
+      />,
+    );
+
+    screen.getByRole('link', { name: 'SecondaryMessage link title' });
```
Comment (MkMan) [inline src/components/SecondaryMessage/SecondaryMessage.spec.tsx:42]: I share Kent's [opinion](https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#using-get-variants-as-assertions) on this but like him I am also not fussed. Up to you 🙂

File: src/components/SecondaryMessage/SecondaryMessage.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import { SecondaryMessage } from './SecondaryMessage';
+import { SecondaryMessageFixture } from './SecondaryMessage.fixture';
+
+describe('<SecondaryMessage />', () => {
```
Comment (MkMan) [inline src/components/SecondaryMessage/SecondaryMessage.spec.tsx:6]: Thanks for improving the tests 🙂

File: src/components/SecondaryMessage/components/Message/Message.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+import { render, screen } from '@testing-library/react';
+import { Message } from './Message';
+import { MessageFixture } from './Message.fixture';
+
+describe('<Message />', () => {
+  it('should present UI with heading copy', () => {
+    render(<Message copy="message copy" heading="heading copy" />);
+
+    expect(
+      screen.getByRole('heading', { name: 'heading copy' }),
```
Comment (MkMan) [inline src/components/SecondaryMessage/components/Message/Message.spec.tsx:11]: I wanted your thoughts on this. Would it be better to mock the Heading and Link and assert that the mocks are called with specific props? Currently, this is testing the implementation of the Heading component (that it renders a `<h>`).

File: src/components/SecondaryMessage/SecondaryMessage.spec.tsx (hunk 1 above)
Comment (beau-charman) [reply src/components/SecondaryMessage/SecondaryMessage.spec.tsx:42]: ah yeah, so do I - thank you for the catch

File: src/components/SecondaryMessage/components/Message/Message.spec.tsx (hunk 3 above)
Comment (beau-charman) [reply src/components/SecondaryMessage/components/Message/Message.spec.tsx:11]: I see what you mean, but I just want to assert that it 'should present UI with heading copy', hence explicitly checking that the resulting ui contains `'heading', { name: 'heading copy' }`

**Review by MkMan — APPROVED** [review]

**Pull request merged**